loadtest/control/simulcontroller/actions.go

```diff
@@ -114,31 +193,160 @@ func switchChannel(u user.User) control.UserActionResponse {
+	// TODO: possibly add some additional idle time here to simulate the
+	// user actually taking time to type a post message.
+	u.SendTypingEvent(channel.Id, "")
+
+	// This is an estimate that comes from stats on community servers.
+	// The average length (in words) for a root post (not a reply).
+	// TODO: should be part of some advanced configuration.
+	avgWordCount := 34
+	minWordCount := 1
+
+	// TODO: make a util function out of this behaviour.
+	wordCount := rand.Intn(avgWordCount*2-minWordCount*2) + minWordCount
+
+	postId, err := u.CreatePost(&model.Post{
+		Message:   control.GenerateRandomSentences(wordCount),
+		ChannelId: channel.Id,
+		CreateAt:  time.Now().Unix() * 1000,
+	})
+	if err != nil {
 		return control.UserActionResponse{Err: control.NewUserError(err)}
 	}
 
-	if err := u.GetPostsSince(channel.Id, time.Now().Add(-1*time.Minute).Unix()*1000); err != nil {
+	return control.UserActionResponse{Info: fmt.Sprintf("post created, id %v", postId)}
+}
+
+func createDirectChannel(u user.User) control.UserActionResponse {
+	// Here we make a call to GetUsers to simulate the user opening the users
+	// list when creating a direct channel.
+	if err := u.GetUsers(0, 100); err != nil {
 		return control.UserActionResponse{Err: control.NewUserError(err)}
 	}
 
-	return control.UserActionResponse{Info: fmt.Sprintf("switched to channel %s", channel.Id)}
+	// TODO: make the selection a bit smarter and pick someone
+	// we don't have a direct channel with already.
+	user, err := u.Store().RandomUser()
+	if errors.Is(err, memstore.ErrLenMismatch) {
+		return control.UserActionResponse{Info: "not enough users to create direct channel"}
+	} else if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	channelId, err := u.CreateDirectChannel(user.Id)
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	if err := u.GetChannel(channelId); err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	if err := u.GetChannelMember(channelId, u.Store().Id()); err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	// We need to update the user's preferences so that
+	// on next reload we can properly fetch opened DMs.
+	pref := &model.Preferences{
+		model.Preference{
+			UserId:   u.Store().Id(),
+			Category: model.PREFERENCE_CATEGORY_DIRECT_CHANNEL_SHOW,
+			Name:     channelId,
+			Value:    "true",
+		},
+	}
+
+	if err := u.UpdatePreferences(pref); err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	channel, err := u.Store().Channel(channelId)
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	if resp := viewChannel(u, channel); resp.Err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	return control.UserActionResponse{Info: fmt.Sprintf("direct channel created, id %s", channelId)}
```

Comment:
Thinking out loud; when I create a direct channel, I go ahead and send a message, to be more simulative maybe we can add a `CreatePost()` event here.

Reply:
That's a good point :+1: